packages/transaction-controller/test/JsonRpcRequestMocks.ts

```diff
@@ -0,0 +1,232 @@
+import type { Hex } from '@metamask/utils';
+
+import type { JsonRpcRequestMock } from '../../../tests/mock-network';
+
+/**
+ * Builds mock eth_gasPrice request.
+ * Used by getSuggestedGasFees.
+ *
+ * @param result - the hex gas price result.
+ * @returns The mock json rpc request object.
+ */
+export function buildEthGasPriceRequestMock(
+  result: Hex = '0x1',
+): JsonRpcRequestMock {
+  return {
+    request: {
+      method: 'eth_gasPrice',
+      params: [],
+    },
+    response: {
+      result,
+    },
+  };
+}
+
+/**
+ * Builds mock eth_blockNumber request.
+ * Used by NetworkController and BlockTracker.
+ *
+ * @param result - the hex block number result.
+ * @returns The mock json rpc request object.
+ */
+export function buildEthBlockNumberRequestMock(
+  result: Hex,
+): JsonRpcRequestMock {
+  return {
+    request: {
+      method: 'eth_blockNumber',
+      params: [],
+    },
+    response: {
+      result,
+    },
+  };
+}
+
+/**
+ * Builds mock eth_getCode request.
+ * Used by readAddressAsContract and requiresFixedGas.
+ *
+ * @param address - The hex address.
+ * @param blockNumber - The hex block number.
+ * @param result - the hex code result.
+ * @returns The mock json rpc request object.
+ */
+export function buildEthGetCodeRequestMock(
+  address: Hex,
+  blockNumber: Hex = '0x1',
+  result: Hex = '0x',
+): JsonRpcRequestMock {
+  return {
+    request: {
+      method: 'eth_getCode',
+      params: [address, blockNumber],
+    },
+    response: {
+      result,
+    },
+  };
+}
+
+/**
+ * Builds mock eth_getBlockByNumber request.
+ * Used by NetworkController.
+ *
+ * @param blockNumber - The hex block number.
+ * @param baseFeePerGas - the hex base fee per gas result.
+ * @param number - the hex (block) number result.
+ * @returns The mock json rpc request object.
+ */
+export function buildEthGetBlockByNumberRequestMock(
+  blockNumber: Hex,
+  baseFeePerGas: Hex = '0x63c498a46',
+  number: Hex = '0x1',
```

Comment:
[nit] why should we have a default for this when there is none for the `blockNumber` arg? Can these just be consolidated into the same arg? Aren't they meant to be the same?